### PR TITLE
Release/0.5.0

### PR DESCRIPTION
**Pull request by ungn**
Factotum "server mode":
- [x] http server
- [x] json
  - [x] parse
  - [x] construct
  - [x] validate
- [x] state
  - ~~log file/database~~
  - [x] consul
- [x] dispatcher
  - [x] one-to-many queue
  - [x] add to queue
  - [x] get queue status
- [x] worker process job
- [x] cli
- [x] ~~merge into main factotum Cargo.toml~~ move to separate repo
- [ ] ci/cd
- [ ] docs

### CODE REVIEW

**Comment by alexanderdean** [comment]
It's coming along @ungn!

File: executor/src/executor/server/dispatcher/mod.rs [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#[derive(Debug)]
+pub struct Dispatcher {
+    job_queue: Vec<JobRequest>,
+    worker_queue: Vec<JobRequest>
+}
+
+impl Dispatcher {
+    pub fn new() -> Dispatcher {
+        Dispatcher {
+            job_queue: Vec::new(),
```
Comment (ninjabear) [inline executor/src/executor/server/dispatcher/mod.rs]: `vec![]`

**Comment by ninjabear** [comment]
Hey @ungn, just to recap my thoughts on what the "thread pool/channels" part of this will look like. 

At the beginning of the process we create a new thread that will be our "worker manager". We can communicate with this thread using a [channel](http://rustbyexample.com/std_misc/channels.html). Each request that comes in should be transformed into a strongly typed request for the "worker manager" by the iron handlers (this means our "worker manager" can be unit tested without being coupled to the actual http requests). 

This "worker manager" is responsible for:

1. ensuring no more than N jobs are running at once (others are queued) 
2. providing information on how many jobs are queued and so on.

Jobs themselves can be dispatched on [futures](https://tokio.rs/docs/getting-started/futures/) provided by [this crate](https://github.com/alexcrichton/futures-rs).

We may not need a separate thread to process a channel (or even need a channel at all). My only reason for including it is so we can dequeue jobs as running jobs are completed (separate from http calls). 

So, a bit more on "seams" and why I always ramble about them, the Iron request handlers are responsible for converting HTTP requests into some kind of request understandable by our business logic. This means that we can test the business logic without iron. We should be able to test the request handlers convert the right kind of data into business logic requests, also without iron. 

What we don't need to do however is spin up iron and see that our requests "really" turn into jobs and so on. It'd be nice to add this level of integration testing - entirely optional. I'm on the "tests should run faster than facebook.com loads" kind of camp. Apologies if I'm teaching you to suck eggs here! 

It's cool if through looking at this in more detail you come up with something better. I just wanted to explain one way that could work - the only bad plan is one that cannot be changed imo.

**Comment by ungn** [comment]
TODO:
- [ ] add default values for cmd line args
- [ ] update app logging:
  - [ ] change from file to stdout/err
  - [ ] add cmd line arg for absolute log file path (if unable to change)
  - [ ] add cmd line arg for log level
- [ ] ignore jobId field in job request when parsing json (should not be required as it is generated)
- [ ] remap rust snake_case fields to json camelCase using serde
- [ ] remove system stats from status message
- [ ] read in request url query string (pretty=1, id params, etc)
- [ ] return better error messages (possible iron framework limitation that needs workaround)

**Comment by ungn** [comment]
Closing as changes for #98 are covered in separate [factotum-server project](https://github.com/snowplow/factotum-server) and #94 is parked until a better solution for underlying problem can be found (see comments for further info).

**Pull request closed without merging**